src/main/kotlin/org/veupathdb/service/eda/compute/EDA.kt

```diff
@@ -0,0 +1,182 @@
+import org.veupathdb.service.eda.compute.service.ServiceOptions
+import org.veupathdb.service.eda.compute.util.JobIDs
+import org.veupathdb.service.eda.compute.util.toAuthTuple
+import org.veupathdb.service.eda.compute.util.toJobResponse
+import org.veupathdb.service.eda.generated.model.APIFilter
+import org.veupathdb.service.eda.generated.model.APIStudyDetail
+import org.veupathdb.service.eda.generated.model.ComputeRequestBase
+import org.veupathdb.service.eda.generated.model.JobResponse
+import java.io.InputStream
+import java.util.*
+
+/**
+ * EDA Project/Service Access Singleton
+ *
+ * Home location for single functions used to work with various EDA services
+ * including the compute service.
+ *
+ *
+ */
+object EDA {
+
+  private val Log = LogManager.getLogger(javaClass)
+
+  /**
+   * Returns the [StudyAccess] permissions object for the target [studyID] and
+   * user described by the given [auth] header.
+   *
+   * @param studyID ID of the study for which the target user's permissions
+   * should be fetched.
+   *
+   * @param auth Auth header sent in with the HTTP request describing the user
+   * whose permissions should be fetched.
+   *
+   * @return `StudyAccess` permissions object describing the permissions the
+   * target user has for the target study.
+   */
+  @JvmStatic
+  fun getStudyPerms(studyID: String, auth: TwoTuple<String, String>): StudyAccess =
+    DatasetAccessClient(ServiceOptions.datasetAccessHost, auth).getStudyAccess(studyID)
+
+  /**
+   * Fetches the [APIStudyDetail] information from the EDA Subsetting Service
+   * for the given [studyID] if such a study exists.
+   *
+   * @param studyID ID of the study whose metadata should be retrieved.
+   *
+   * @param auth Auth header sent in with the job HTTP request.
+   *
+   * @return An [Optional] that will wrap the `APIStudyDetail` returned from the
+   * EDA Subsetting Service, if such a study exists.  If the target study was
+   * not found, the returned `Optional` will be empty.
+   */
+  @JvmStatic
+  fun getAPIStudyDetail(studyID: String, auth: TwoTuple<String, String>): Optional<APIStudyDetail> =
+    EdaSubsettingClient(ServiceOptions.edaSubsettingHost, auth).getStudy(studyID)
+
+  /**
+   * Fetches the [APIStudyDetail] information from the EDA Subsetting Service
+   * for the given [studyID], throwing an exception if no such study exists.
+   *
+   * @param studyID ID of the study whose metadata should be retrieved.
+   *
+   * @param auth Auth header sent in with the job HTTP request.
+   *
+   * @param err Optional exception provider that will be used to get the
+   * exception that will be thrown if the target study does not exist.
+   *
+   * @return The `APIStudyDetail` information returned from the EDA Subsetting
+   * Service.
+   *
+   * @throws Exception Throws the exception returned by the given exception
+   * provider ([err]).
+   */
+  @JvmStatic
+  @JvmOverloads
+  fun requireAPIStudyDetail(
```

Comment:
why is this method called `requireAPIStudyDetail`?  why "require"?

Reply:
Because the standard method (`getAPIStudyDetail`) may return `null` and this one requires that the value is present, throwing an exception if not rather than returning `null`.

It's a fairly common pattern elsewhere to have methods prefixed with "must" or "require" when those methods exist alongside similar methods that are more loose in their allowed behavior or return types.